### ISSUE TITLE
freeze no effect

### ISSUE THREAD

**Issue by sjz-hub** [ultralytics/ultralytics#9391]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I tried to train a model using yolov8n and used free=22 to lock its backbone, but in the end, I extracted the backbone from the trained model as a separate model. And use its backbone with yolov8n.pt to input the same image, but the output tensor results are not the same. Then I compared the contents of the two backbones and found that most of them were the same, but the contents of model.4. m.1. cv1. bn. running_var and model.4. m.1. cv1. bn. running_mea are different. How can I lock a backbone so that all its parameters remain unchanged for training

**Comment by glenn-jocher** [comment]
@sjz-hub hello! Thanks for reaching out and sharing your findings. It seems like you're attempting to freeze the backbone layers of YOLOv8n to prevent them from updating during training. To achieve this, the correct argument to pass is `freeze`, not `free`. Based on your description, you should use `freeze=22` correctly in your training command. Here's an example:

```bash
yolo detect train data=coco128.yaml model=yolov8n.yaml epochs=100 imgsz=640 freeze=22
```

This command should freeze the first 22 layers of the YOLOv8n model during training. 

However, it's important to note that while `freeze` prevents the weights of these layers from being updated, the batch normalization layers' `running_mean` and `running_var` might still update during training. This behavior is by design to keep batch normalization layers adapting to the data they see.

If you need to freeze these as well, you might need to modify the source code to prevent updates to these parameters or explore using a custom training loop that explicitly handles this case.

I hope this helps! If you have any more questions, feel free to ask. 🌟
